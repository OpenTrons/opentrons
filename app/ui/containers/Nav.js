// top-level container
import React from 'react'
import {connect} from 'react-redux'

import {
  actions as interfaceActions,
  selectors as interfaceSelectors
} from '../interface'

import {
  actions as robotActions,
  selectors as robotSelectors
} from '../robot'

import uploadIconSrc from '../img/icon_file.svg'
import setupIconSrc from '../img/icon_setup.svg'

import SideBar from '../components/SideBar'

const mapStateToProps = (state) => {
  const isConnected = state.robot.isConnected
  return {
    // interface
    isNavPanelOpen: interfaceSelectors.getIsNavPanelOpen(state),
    currentNavPanelTask: interfaceSelectors.getCurrentNavPanelTask(state),
    navLinks: [
      {name: 'upload', iconSrc: uploadIconSrc, isDisabled: !isConnected, to: '/upload'},
      {name: 'setup', iconSrc: setupIconSrc, isDisabled: !isConnected, to: '/setup-instruments'}
    ],

    // robot
    isReadyToRun: robotSelectors.getIsReadyToRun(state),
    isRunning: robotSelectors.getIsRunning(state),
    connectionStatus: robotSelectors.getConnectionStatus(state),
    isConnected
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    // interface
    onNavClick: () => dispatch(interfaceActions.toggleNavPanel()),
    onNavIconClick: (panel) => () => {
      dispatch(interfaceActions.setCurrentNavPanel(panel))
    },

    // robot
    // TODO(mc): revisit when robot discovery / multiple robots is addressed
    onConnectClick: () => dispatch(robotActions.connect()),
    onDisconnectClick: () => dispatch(robotActions.disconnect()),
    onRunClick: () => dispatch(robotActions.run()),

    // session
    onUpload: (event) => dispatch(robotActions.session(event.target.files[0]))
  }
}

function Nav (props) {
  return (
    <SideBar {...props} />
  )
}

export default connect(mapStateToProps, mapDispatchToProps)(Nav)
