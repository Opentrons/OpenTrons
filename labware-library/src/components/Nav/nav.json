{
  "homeUrl": "https://opentrons.com",
  "subdomainList": [
    {
      "name": "Protocol Designer",
      "url": "https://designer.opentrons.com/",
      "linkout": true,
      "gtm": {
        "action": "click",
        "category": "toolbar",
        "label": "protocol-designer"
      }
    },
    {
      "name": "Protocol Library",
      "url": "https://protocols.opentrons.com/",
      "linkout": true,
      "gtm": {
        "action": "click",
        "category": "toolbar",
        "label": "protocol-library"
      }
    },
    {
      "name": "Python API",
      "url": "https://docs.opentrons.com/",
      "linkout": true,
      "gtm": {
        "action": "click",
        "category": "toolbar",
        "label": "opentrons-api"
      }
    },
    {
      "name": "Labware Library",
      "url": "https://labware.opentrons.com/",
      "linkout": true,
      "gtm": {
        "action": "click",
        "category": "toolbar",
        "label": "labware-library"
      }
    },
    {
      "name": "GitHub",
      "url": "https://github.com/Opentrons",
      "linkout": true,
      "gtm": {
        "action": "click",
        "category": "toolbar",
        "label": "github-protocols"
      }
    },
    {
      "name": "Opentrons App",
      "url": "/ot-app",
      "linkout": false,
      "gtm": {
        "action": "click",
        "category": "toolbar",
        "label": "opentrons-app"
      }
    }
  ],
  "navigationList": [
    {
      "name": "About",
      "links": [
        {
          "links": [
            {
              "name": "Mission",
              "url": "/about",
              "linkout": false,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "about"
              }
            },
            {
              "name": "Our Customers",
              "url": "/customers",
              "linkout": false,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "customers"
              }
            },
            {
              "name": "Press",
              "url": "http://blog.opentrons.com/press",
              "linkout": true,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "press"
              }
            },
            {
              "name": "Our Team",
              "url": "/team",
              "linkout": false,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "team"
              }
            },
            {
              "name": "How it works",
              "url": "/how-it-works/",
              "linkout": false,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "how-it-works"
              }
            }
          ]
        }
      ]
    },
    {
      "name": "Products",
      "text": "100% transparent pricing. <a href=\"https://shop.opentrons.com\" target=\"_blank\" rel=\"noopener noreferrer\" data-gtm-category=\"b-header\" data-gtm-label=\"all-products\" data-gtm-action=\"click\">Order online  ›</a>",
      "links": [
        {
          "name": "OT-2 LAB ROBOT",
          "links": [
            {
              "name": "OT-2 Robot",
              "url": "/ot-2",
              "linkout": false,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "ot-2"
              }
            },
            {
              "name": "OT-2 Pipettes",
              "url": "/pipettes",
              "linkout": false,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "ot-2-pipettes"
              }
            }
          ]
        },
        {
          "name": "OT-2 WORKSTATIONS",
          "links": [
            {
              "name": "Nucleic Acid Cleanup",
              "linkout": false,
              "url": "/workstations/automated-nucleic-acid-cleanup",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "workstation-nac"
              }
            },
            {
              "name": "NGS Library Prep",
              "url": "/workstations/automated-ngs-library-prep/",
              "linkout": false,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "ngs-prep"
              }
            }
          ]
        },
        {
          "name": "OT-2 HARDWARE MODULES",
          "links": [
            {
              "name": "Thermocycler Module",
              "url": "/modules/thermocycler-module",
              "linkout": false,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "thermocycler"
              }
            },
            {
              "name": "Temperature Module",
              "url": "/modules/temperature-module",
              "linkout": false,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "tempmod"
              }
            },
            {
              "name": "Magnetic Module",
              "url": "/modules/magnetic-module",
              "linkout": false,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "magmod"
              }
            }
          ]
        },
        {
          "name": "CONSUMABLES & REAGENTS",
          "links": [
            {
              "name": "Tips & Filter Tips",
              "linkout": true,
              "url": "https://shop.opentrons.com/collections/opentrons-tips",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "tips"
              }
            },
            {
              "name": "Reagents",
              "linkout": true,
              "url": "https://shop.opentrons.com/collections/verified-reagents",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "reagents"
              }
            },
            {
              "name": "Tubes & Vials",
              "linkout": true,
              "url": "https://shop.opentrons.com/collections/tubes",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "tube"
              }
            }
          ]
        },
        {
          "name": "VERIFIED LABWARE",
          "links": [
            {
              "name": "Lab Plates",
              "linkout": true,
              "url": "https://shop.opentrons.com/collections/lab-plates",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "plates"
              }
            },
            {
              "name": "Reservoirs",
              "linkout": true,
              "url": "https://shop.opentrons.com/collections/reservoirs",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "reserv"
              }
            },
            {
              "name": "Tube Racks",
              "linkout": true,
              "url": "https://shop.opentrons.com/collections/racks-and-adapters",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "tubes-racks"
              }
            }
          ]
        }
      ]
    },
    {
      "name": "Applications",
      "url": "/automated-lab-applications/",
      "linkout": false,
      "gtm": {
        "action": "click",
        "category": "header",
        "label": "applications"
      }
    },
    {
      "name": "Blog",
      "url": "https://blog.opentrons.com",
      "linkout": true,
      "gtm": {
        "action": "click",
        "category": "header",
        "label": "blog"
      }
    },
    {
      "name": "Support",
      "text": "<a href=\"https://opentrons.com/contact-support/\"  target=\"_blank\" rel=\"noopener noreferrer\" data-gtm-category=\"b-header\" data-gtm-label=\"contact-support\" data-gtm-action=\"click\">Contact support  ›</a>",
      "links": [
        {
          "links": [
            {
              "name": "Product Help Center",
              "linkout": true,
              "url": "https://support.opentrons.com/",
              "description": "Explore our in-depth support articles",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "product-help"
              }
            },
            {
              "name": "OT-2 Start guide",
              "linkout": true,
              "url": "https://support.opentrons.com/guide-for-getting-started-with-the-ot-2",
              "description": "You recieved your robot, here's what's next",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "start-guide"
              }
            }
          ]
        },
        {
          "links": [
            {
              "name": "Python API",
              "linkout": true,
              "url": "https://docs.opentrons.com/",
              "description": "Easily write new protocols using Python",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "opentrons-api"
              }
            },
            {
              "name": "Documentation",
              "url": "https://blog.opentrons.com/resources/",
              "description": "Manuals, white papers, eBooks, app notes, etc.",
              "linkout": true,
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "documentation"
              }
            },
            {
              "name": "Return Policy",
              "linkout": true,
              "url": "https://support.opentrons.com/en/articles/3670461-our-risk-free-return-policy",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "return-policy"
              }
            },
            {
              "name": "Warranty",
              "linkout": true,
              "url": "https://support.opentrons.com/en/articles/736748-what-is-your-warranty-and-return-policy",
              "gtm": {
                "action": "click",
                "category": "header",
                "label": "warranty"
              }
            }
          ]
        }
      ]
    }
  ]
}
