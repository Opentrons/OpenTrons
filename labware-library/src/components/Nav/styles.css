/* top navbar styles */
@import '@opentrons/components';
@import '../../styles/breakpoints.css';
@import '../../styles/shadows.css';
@import '../../styles/spacing.css';

:root {
  --subdomain-nav-max-width: 78.75rem;
  --subdomain-nav-padding: 1.25rem;
  --shadow-1: 0 2px 2px rgba(0, 0, 0, 0.2);
  --nav-border: 1px solid #e3e3e3;
}

.nav {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  z-index: 999;
}

.subdomain_nav_wrapper {
  display: none;

  @media (--large) {
    display: block;
    width: 100%;
    height: 30px;
    background-color: var(--c-white);
    border-bottom: var(--nav-border);
  }
}

.nav_container {
  width: 100%;
  height: 100%;
  padding: var(--spacing-1) var(--spacing-6);

  @media (--medium) {
    padding-left: var(--subdomain-nav-padding);
    padding-right: var(--subdomain-nav-padding);
  }

  @media (--large) {
    margin: auto;
    max-width: var(--screen-width-xlarge);
    padding: 0 40px;
  }
}

.main_nav_wrapper {
  width: 100%;
  height: var(--size-mobile-nav);
  background-color: var(--c-white);
  border-bottom: var(--bd-light);

  @media (--medium) {
    height: var(--size-main-nav);
  }

  @media (--large) {
    border: none;
    box-shadow: var(--shadow-1);
  }
}

.breadcrumbs {
  position: fixed;
  top: var(--size-mobile-nav);
  left: 0;
  right: 0;
  z-index: 888;
  width: 100%;
  font-size: var(--fs-default);
  font-weight: var(--fw-semibold);
  line-height: var(--lh-copy);
  background-color: var(--c-lightest-gray);
}

.breadcrumbs_contents {
  display: flex;
  align-items: center;
  height: var(--size-2);
  padding: 0 var(--spacing-5);
  overflow: visible;
}

.breadcrumbs_link {
  @apply --transition-color;

  color: var(--c-blue);
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  &:hover,
  &:active {
    color: color-mod(var(--c-blue) alpha(0.6));
  }
}

.breadcrumbs_separator {
  flex: none;
  margin: 0 var(--spacing-2);
}

@media (--medium) {
  .breadcrumbs {
    top: var(--size-main-nav);
  }

  .breadcrumbs_contents {
    padding-left: var(--spacing-7);
    padding-right: var(--spacing-7);
  }
}

@media (--large) {
  .breadcrumbs {
    top: var(--size-total-nav);
  }

  .breadcrumbs_contents {
    margin: auto;
    max-width: var(--screen-width-xlarge);
  }
}
