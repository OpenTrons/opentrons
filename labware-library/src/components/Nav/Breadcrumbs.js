// @flow
import * as React from 'react'
import { Link } from 'react-router-dom'

import { getPublicPath } from '../../public-path'
import styles from './styles.css'

import type { LabwareDefinition } from '../../types'

// TODO(mc, 2019-04-07): i18n
const EN_LABWARE_LIBRARY = 'Labware Library'

export type BreadcrumbsProps = {
  definition: LabwareDefinition | null,
}

export default function Breadcrumbs(props: BreadcrumbsProps) {
  const { definition } = props
  if (!definition) return null

  return (
    <div className={styles.breadcrumbs}>
      <div className={styles.breadcrumbs_contents}>
        <Link to={getPublicPath()} className={styles.breadcrumbs_link}>
          {EN_LABWARE_LIBRARY}
        </Link>
        <span className={styles.breadcrumbs_separator}>{' > '}</span>
        <Link
          to={`${getPublicPath()}${definition.parameters.loadName}`}
          className={styles.breadcrumbs_link}
        >
          {definition.metadata.displayName}
        </Link>
      </div>
    </div>
  )
}
