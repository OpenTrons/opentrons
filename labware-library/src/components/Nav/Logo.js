// @flow
// top nav bar logo image
import * as React from 'react'

import logoSrc from './images/ot-logo-full.png'
import styles from './styles.css'

export default function Nav () {
  return <img className={styles.logo} src={logoSrc} />
}
