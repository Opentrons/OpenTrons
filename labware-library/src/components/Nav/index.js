// @flow
// top nav bar component
import * as React from 'react'
import { SubdomainNav } from '../SubdomainNav'
import { MainNav } from '../MainNav'
import styles from './styles.css'
import navData from './nav.json'

export { Breadcrumbs } from './Breadcrumbs'

export function Nav(): React.Node {
  const { navigationList, subdomainList, homeUrl: url } = navData
  const [navigation, setNavigation] = React.useState(navigationList)
  const [homeUrl, setHomeUrl] = React.useState(url)
  const [subdomain, setSubdomain] = React.useState(subdomainList)

  React.useEffect(() => {
    window
      .fetch('https://opentrons.com/api/navigationData.json')
      .then(response => {
        response.json().then(data => {
          if (data && data.homeUrl) {
            setSubdomain(data.subdomainList)
            setNavigation(data.navigationList)
            setHomeUrl(data.homeUrl)
          }
        })
      })
      .catch(e => console.error(e))
  }, [])

  return (
    <>
      <nav className={styles.nav}>
        <div className={styles.subdomain_nav_wrapper}>
          <div className={styles.nav_container}>
            <SubdomainNav subdomainList={subdomain} homeUrl={homeUrl} />
          </div>
        </div>
        <div className={styles.main_nav_wrapper}>
          <div className={styles.nav_container}>
            <MainNav navigationList={navigation} homeUrl={homeUrl} />
          </div>
        </div>
      </nav>
    </>
  )
}
