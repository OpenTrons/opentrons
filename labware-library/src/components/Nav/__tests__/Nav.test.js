// @flow
// tests for top navbar
import * as React from 'react'
import Renderer from 'react-test-renderer'

import Nav from '..'

describe('Nav', () => {
  test('component renders', () => {
    const tree = Renderer.create(<Nav />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
