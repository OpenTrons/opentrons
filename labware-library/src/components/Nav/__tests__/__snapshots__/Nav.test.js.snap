// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders 1`] = `
<Fragment>
  <nav
    className="nav"
  >
    <div
      className="subdomain_nav_wrapper"
    >
      <div
        className="nav_container"
      >
        <SubdomainNav />
      </div>
    </div>
    <div
      className="main_nav_wrapper"
    >
      <div
        className="nav_container"
      >
        <MainNav
          isMobileOpen={false}
          onMobileClick={[Function]}
        />
      </div>
    </div>
  </nav>
</Fragment>
`;
